sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_replica_client.py

```diff
@@ -0,0 +1,327 @@
+        :param str user_agent: The user agent string to use for the request
+        :param int retry_total: The total number of retries to allow for requests
+        :param int retry_backoff_max: The maximum backoff time for retries
+        :return: A new instance of the ReplicaClient class
+        :rtype: ReplicaClient
+        """
+        return cls(
+            endpoint,
+            AzureAppConfigurationClient(
+                endpoint,
+                credential,
+                user_agent=user_agent,
+                retry_total=retry_total,
+                retry_backoff_max=retry_backoff_max,
+                **kwargs
+            ),
+        )
+
+    @classmethod
+    def from_connection_string(
+        cls, endpoint: str, connection_string: str, user_agent: str, retry_total: int, retry_backoff_max: int, **kwargs
+    ) -> "ReplicaClient":
+        """
+        Creates a new instance of the ReplicaClient class, using the provided connection string to authenticate
+        requests.
+
+        :param str endpoint: The endpoint of the App Configuration store
+        :param str connection_string: The connection string to use for authentication
+        :param str user_agent: The user agent string to use for the request
+        :param int retry_total: The total number of retries to allow for requests
+        :param int retry_backoff_max: The maximum backoff time for retries
+        :return: A new instance of the ReplicaClient class
+        :rtype: ReplicaClient
+        """
+        return cls(
+            endpoint,
+            AzureAppConfigurationClient.from_connection_string(
+                connection_string,
+                user_agent=user_agent,
+                retry_total=retry_total,
+                retry_backoff_max=retry_backoff_max,
+                **kwargs
+            ),
+        )
+
+    def _check_configuration_setting(
+        self, key: str, label: str, etag: Optional[str], headers: Dict[str, str], **kwargs
+    ) -> Tuple[bool, Union[ConfigurationSetting, None]]:
+        """
+        Checks if the configuration setting have been updated since the last refresh.
+
+        :param str key: key to check for chances
+        :param str label: label to check for changes
+        :param str etag: etag to check for changes
+        :param Mapping[str, str] headers: headers to use for the request
+        :return: A tuple with the first item being true/false if a change is detected. The second item is the updated
+        value if a change was detected.
+        :rtype: Tuple[bool, Union[ConfigurationSetting, None]]
+        """
+        try:
+            updated_sentinel = self._client.get_configuration_setting(  # type: ignore
+                key=key, label=label, etag=etag, match_condition=MatchConditions.IfModified, headers=headers, **kwargs
+            )
+            if updated_sentinel is not None:
+                logging.debug(
```

Comment:
You should not call the root logger's `debug` - provide a module level Logger instance instead.